package-schema-registry/test/test_schema_registry.py

```diff
@@ -0,0 +1,354 @@
+import os
+import unittest
+import time
+import string
+import json
+
+import confluent.docker_utils as utils
+
+CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
+FIXTURES_DIR = os.path.join(CURRENT_DIR, "fixtures")
+KAFKA_READY = "bash -c 'cub kafka-ready {brokers} 40 -z $KAFKA_ZOOKEEPER_CONNECT && echo PASS || echo FAIL'"
+HEALTH_CHECK = "bash -c 'cub sr-ready {host} {port} 20 && echo PASS || echo FAIL'"
+POST_SCHEMA_CHECK = """curl -X POST -i -H "Content-Type: application/vnd.schemaregistry.v1+json" \
+    --data '{"schema": "{\\"type\\": \\"string\\"}"}' \
+    %s:%s/subjects/%s/versions"""
+GET_SCHEMAS_CHECK = "bash -c 'curl -X GET -i {host}:{port}/subjects'"
+ZK_READY = "bash -c 'cub zk-ready {servers} 40 && echo PASS || echo FAIL'"
+KAFKA_CHECK = "bash -c 'kafkacat -L -b {host}:{port} -J' "
+
+JMX_CHECK = """bash -c "\
+    echo 'get -b kafka.schema.registry:type=jetty-metrics connections-active' |
+        java -jar jmxterm-1.0-alpha-4-uber.jar -l {jmx_hostname}:{jmx_port} -n -v silent "
+"""
+
+
+class ConfigTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
```

Comment:
are we doing this just to save time on pulling the cluster up/down for each test? just asking because we've found for system tests that trying to do stuff like that ends up being error prone, tests end up depending on test ordering, etc.

Reply:
These tests were pulled out of the old repo, but I think that was the idea.

Several of the tests follow this same pattern -- and sure enough, they are proving to be flakey. All of these tests need love (tough love?) from the experts.